### PR TITLE
Added FairSeq checkpointing tests, to local disk and gcs.

### PR DESCRIPTION
**Pull request by taylanbil**

### CODE REVIEW

**Comment by taylanbil** [comment]
[oneshot](https://pantheon.corp.google.com/kubernetes/job/us-central1-b/xl-ml-test/default/pt-nightly-fs-checkpoint-gcs-functional-v3-8-bhk5k?project=xl-ml-test&tab=details&duration=P30D&pod_summary_list_tablesize=20&service_list_datatablesize=20) submitted.

File: templates/pytorch/nightly/fs-transformer.libsonnet [hunk 1]
```diff
@@ -52,76 +87,47 @@ local utils = import "../../utils.libsonnet";
       },
     },
   },
+  local checkpoint_local = base.Functional {
+    modelName: "fs-checkpoint-local",
+    command: utils.scriptCommand(
+      |||
+        %(common)s  --max-epoch=1 \
+          --save-dir=/tmp/checkpoints
+        %(common)s  --max-epoch=2 \
```
Comment (taylanbil) [inline templates/pytorch/nightly/fs-transformer.libsonnet:96]: the goal here is to test loading/resuming from the chpt saved by the previous command.

**Comment by will-cromar** [comment]
I just checked and your oneshot run timed out. The code looks fine, but can you double check that this passes before submitting?

**Comment by taylanbil** [comment]
[oneshot](https://pantheon.corp.google.com/logs/viewer?interval=NO_LIMIT&project=xl-ml-test&minLogLevel=0&expandAll=false&timestamp=2020-04-07T18:49:51.051000000Z&customFacets=&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22xl-ml-test%22%0Aresource.labels.location%3D%22us-central1-b%22%0Aresource.labels.cluster_name%3D%22oneshots%22%0Aresource.labels.namespace_name%3D%22default%22%0Alabels.k8s-pod%2Fcontroller-uid%3D%22be350923-717e-446e-bdfe-15e3c7dbc299%22&scrollTimestamp=2020-04-07T17:43:52.284739160Z) that passed.

**Comment by zcain117** [comment]
This is really cool, awesome to see such flexible use of this new testing framework to get some new types of test coverage 💯

**Pull request merged**